ddtrace/contrib/internal/google_adk/patch.py

```diff
@@ -34,7 +34,9 @@ def get_version() -> str:
 def _traced_agent_run_async(adk, pin, wrapped, instance, args, kwargs):
     """Trace the main execution of an agent (async generator)."""
     integration: GoogleAdkIntegration = adk._datadog_integration
-    provider_name, model_name = extract_provider_and_model_name(instance=instance.agent.model, model_name_attr="model")
+    provider_name, model_name = extract_provider_and_model_name(
+        instance=getattr(getattr(instance, "agent", {}), "model", {}), model_name_attr="model"
+    )
```

Comment:
```suggestion
    agent = getattr(instance, "agent", None)
    model = getattr(agent, "model", None)
    provider_name, model_name = extract_provider_and_model_name(
        instance=model, model_name_attr="model"
    )
```

i always find it easier to debug future escalations with separating it out like this (and it looks like `extract_provider_and_model_name` does handle `None` properly). we can do this in the other spots too!